cli/run.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2020. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import os
+import time
+import sys
+import signal
+import subprocess
+import webbrowser
+from contextlib import contextmanager
+
+import click
+
+
+@contextmanager
+def kill_process_group_afterwards():
+    os.setpgrp()
+    try:
+        yield
+    finally:
+        # Kill all processes in my group
+        os.killpg(0, signal.SIGKILL)
+
+
+@click.command(
+    name="run",
+    help="Run an experiment on a scenario",
+    context_settings=dict(ignore_unknown_options=True),
+)
+@click.option(
+    "--envision",
+    is_flag=True,
+    default=False,
+    help="Start up Envision server when running an experiment",
+)
+@click.argument(
+    "script_path", type=click.Path(exists=True), metavar="<script>", required=True
+)
+@click.argument("script_args", nargs=-1, type=click.UNPROCESSED)
+def run_experiment(envision, script_path, script_args):
+    with kill_process_group_afterwards():
+        if envision:
+            subprocess.Popen(
+                ["scl", "envision", "start", "-s", "./scenarios", "-p", "8081"],
```

Comment:
I'm not sure how hard this is to do with "click", but if it's not hard, it would be nice to allow specifying (over-riding) the envision port too.  It could be like:
`scl --envision=8082 run ...`
(And if no `=8082` is specified, it just uses the default port.)

Reply:
Sure, I will add that.

Reply:
I don't think its possible to give one option two different types (Boolean and INT).  In fact after some research I think its not possible with Click. So I just added another option `--port` so that users can mention the port they want to run the server on.

Reply:
Ok, yeah that's what I was afraid of.  From a skim of the `click` documentation, it _appears_ like something like this _should_ work for what you were trying to do originally:
```python
@click.option("--envision", type=int, is_flag=False, flag_value=8081, default=None)
```
... but I just tried it and it annoyingly doesn't.

So I guess adding the `--port` option is ok, although I think it's currently not clear that this is _only_ for Envision and has no meaning if used without `--envision`.   So I might call it something like `--envision-port` instead of just `--port`.  And also warn (or fail) if it's specified without specifying `--envision`.

Reply:
Yeah, I had already tried that, but it didn't seem to work.
I will add the port changes. Thanks!